streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/SampleProcessor.kt

```diff
@@ -0,0 +1,73 @@
+package com.mirego.trikot.streams.reactive.processors
+
+import com.mirego.trikot.foundation.FoundationConfiguration
+import com.mirego.trikot.foundation.concurrent.AtomicReference
+import com.mirego.trikot.foundation.timers.TimerFactory
+import com.mirego.trikot.streams.cancellable.CancellableManagerProvider
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+import kotlin.time.Duration
+
+class SampleProcessor<T>(
+    parentPublisher: Publisher<T>,
+    private val interval: Duration,
+    private val timerFactory: TimerFactory = FoundationConfiguration.timerFactory
+) : AbstractProcessor<T, T>(parentPublisher) {
+
+    override fun createSubscription(subscriber: Subscriber<in T>): ProcessorSubscription<T, T> {
+        return SampleProcessorSubscription(subscriber, interval, timerFactory)
+    }
+
+    class SampleProcessorSubscription<T>(
+        private val subscriber: Subscriber<in T>,
+        private val interval: Duration,
+        private val timerFactory: TimerFactory
+    ) : ProcessorSubscription<T, T>(subscriber) {
+        private val cancellableManagerProvider = CancellableManagerProvider()
+        private val lastEmittedValue = AtomicReference<T?>(null)
+
+        private fun initializeTimer() {
+            val cancellableManager = cancellableManagerProvider.cancelPreviousAndCreate()
+
+            val recurringTimer = timerFactory.repeatable(interval) {
+                dispatchSampledValue()
+            }
+
+            cancellableManager.add { recurringTimer.cancel() }
+        }
+
+        override fun onSubscribe(s: Subscription) {
+            super.onSubscribe(s)
+            initializeTimer()
```

Comment:
I feel like we should verify that we're not cancelled or completed here before initializing timer. 

Also, do we really want to cancel and start a new timer every time there is a new subscription?

Reply:
We must start a timer to every new subscription as every subscription is "independent". This is not a shared operator.

Reply:
You're right! Thanks for the clarification. I was under the impression the operator was keeping a state between subscriptions.